File: tools/follows.py
#!/usr/bin/env python3
"""List the accounts a Bluesky user follows (walks the follow graph outward).

    python3 follows.py alice.bsky.social --limit 50
"""
import argparse
import sys

from actor_index import print_actors
from bsky import BskyError, get


def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("actor", help="handle or DID whose follows to list")
    parser.add_argument("--limit", type=int, default=50)
    args = parser.parse_args()
    data = get("app.bsky.graph.getFollows", {"actor": args.actor, "limit": args.limit})
    print_actors(data.get("follows", []))


if __name__ == "__main__":
    try:
        main()
    except BskyError as e:
        print(f"follows fetch failed: {e}", file=sys.stderr)
        sys.exit(1)

File: agent/.opencode/tools/follows.ts
import { tool } from "@opencode-ai/plugin"
import { $ } from "bun"

// Thin wrapper around tools/follows.py — located via the git worktree.
export default tool({
  description:
    "List the accounts a Bluesky user follows (walks the follow graph " +
    "outward). Returns one line per account: `@handle (Name) — did · bio`. " +
    "Use it to see who someone follows, then explore those accounts.",
  args: {
    actor: tool.schema.string().describe("Handle or DID whose follows to list"),
    limit: tool.schema.number().describe("Max accounts to return").default(50),
  },
  async execute(args, context) {
    const script = `${context.worktree}/tools/follows.py`
    const limit = Number.isFinite(args.limit) ? Math.trunc(args.limit as number) : 50
    return await $`python3 ${script} ${args.actor} --limit ${String(limit)}`.env({ ...process.env, ACTING_SESSION: context.sessionID }).text()
  },
})
